go-controller/pkg/util/net_linux.go

```diff
@@ -493,6 +494,22 @@ func GetNetworkInterfaceIPs(iface string) ([]*net.IPNet, error) {
 	return ips, nil
 }
 
+func isAddressReservedForInternalUse(addr net.IP) bool {
```

Comment:
minor nit since GetNetworkInterfaceIPs is a very generic name, it would be good to put a comment in the function text stating Ignores IP addresses that are link local or internally used IPs